code/modules/mob/living/carbon/inventory.dm

```diff
@@ -175,6 +175,10 @@
 
 	update_equipment_speed_mods()
 	hud_used?.update_locked_slots()
+	if(!(slot & item.slot_flags)) // Things below only update if slotted in (ie: not held)
+		return
+	if(item.hair_mask)
+		update_body()
```

Comment:
Can't this be handled in items' own equipped() call rather than here? Could probably even move the var up from ``/obj/item`` to ``/obj/item/clothing/head`` in that case, and put the update call there

Reply:
We can't move it up because any item can be put on a head